staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -106,7 +106,12 @@ func lookupServiceName(name string) (EgressType, error) {
 }
 
 func tunnelHTTPConnect(proxyConn net.Conn, proxyAddress, addr string) (net.Conn, error) {
-	fmt.Fprintf(proxyConn, "CONNECT %s HTTP/1.1\r\nHost: %s\r\n\r\n", addr, "127.0.0.1")
+	_, err := fmt.Fprintf(proxyConn, "CONNECT %s HTTP/1.1\r\nHost: %s\r\n\r\n", addr, addr)
+	if err != nil {
+		return nil, fmt.Errorf("sending HTTP CONNECT request to %s via proxy %s failed: %v",
```

Comment:
I don't think Fprint flushes (?), so it's not clear that this error checking is sufficient.

Reply:
Later in the code we check for resp.StatusCode == 200, so I think this is ok.